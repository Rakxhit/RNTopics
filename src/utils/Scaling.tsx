import {Dimensions, PixelRatio} from 'react-native';

export const normalizeModerately = (size: number, factor = 0.5): number => {
  return PixelRatio.roundToNearestPixel(size, factor);
};

export const normalizeWidth = (size: number): number => {
  return PixelRatio.roundToNearestPixel(size);
};

export const normalizeHeight = (size: number): number => {
  return PixelRatio.roundToNearestPixel(size);
};

export const screenWidth: number = Dimensions.get('window').width;
export const screenHeight: number = Dimensions.get('window').height;
