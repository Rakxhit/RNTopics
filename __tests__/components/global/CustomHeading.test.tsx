import CustomHeading from '../../../src/components/global/CustomHeading';
import {fireEvent, render} from '@testing-library/react-native';
import {goBack} from '../../../src/utils/NavigationUtil';
jest.mock('../../../src/utils/NavigationUtil', () => ({
  goBack: jest.fn(),
}));
describe('Customheader Component', () => {
  it('should render the given text correctly', () => {
    const title = 'HEY THERE';
    const {getByText} = render(<CustomHeading title={title} />);
    expect(getByText(title)).toBeTruthy();
  });

  it('should handle the press event ', () => {
    const {getByTestId} = render(<CustomHeading title="Test" />);
    const backButton = getByTestId('back-button');
    fireEvent.press(backButton);
    expect(goBack).toHaveBeenCalled();
  });

  it('should match the snapshot', () => {
    const {toJSON} = render(<CustomHeading title="Test" />);
    expect(toJSON()).toMatchSnapshot();
  });
});
