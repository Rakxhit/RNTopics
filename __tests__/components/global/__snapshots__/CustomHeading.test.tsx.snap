// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customheader Component should match the snapshot 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="back-button"
  />
  <Text
    style={
      {
        "color": "#fff",
        "fontFamily": "Poppins-SemiBold",
        "fontSize": 22,
        "marginLeft": 8,
        "marginTop": 20,
      }
    }
  >
    Test
  </Text>
</View>
`;
